src/main/java/umk/mat/pajda/ProjektZespolowy/services/TipService.java

```diff
@@ -95,4 +101,26 @@ public ResponseEntity<String> createPayment(OpinionPostDTO opinionPostDTO, Strin
   public void setRestTemplate(RestTemplate restTemplate) {
     this.restTemplate = restTemplate;
   }
+
+  public TipStatisticsGetDTO getStatistics(String userName) {
+    try {
+      User user = userRepository.findByMail(userName).get();
+      TipStatisticsGetDTO tipStatisticsGetDTO = new TipStatisticsGetDTO();
+      Tip tipMin = tipRepository.findFirstByUserOrderByAmountDesc(user);
+      tipStatisticsGetDTO.setMaxTipAmount(tipMin.getAmount());
+      tipStatisticsGetDTO.setCurrency(tipMin.getCurrency());
+
+      tipStatisticsGetDTO.setMinTipAmount(
+          tipRepository.findFirstByUserOrderByAmountAsc(user).getAmount());
+      tipStatisticsGetDTO.setAvgTipAmount(tipRepository.getAvgAmountForAllTips(user.getId()));
+      tipStatisticsGetDTO.setSumTipValueForEveryMonth(
+          tipRepository.getSumAmountForEachMonth(user.getId()));
```

Comment:
i get error for user with id 2

Reply:
fixed

Reply:
![image](https://github.com/Programowanie-Zespolowe-PAJDA/Backend/assets/132700662/e3d4a097-03f5-493b-b44c-7b5d165c538b)
In a given month, should the sum together regardless of currency?

Reply:
resolved , implementation change